### PR TITLE
[WIP] Begin to add some contributing docs (#95)

### PR DESCRIPTION
**Pull request by DJMcNab**
Currently adds some documentation about `cargo gen-kinds`.

Note that I am unsure of some of the details, such as why `DOT` is in `multi_byte_tokens`, hence the [WIP] tag.

Based on #95.

### CODE REVIEW

File: crates/ra_syntax/src/grammar.ron [hunk 1]
```diff
@@ -23,8 +25,9 @@ Grammar(
         ["^", "CARET"],
         ["%", "PERCENT"],
     ],
+    // TODO: Confirm surmision: the tokens which cannot be recorded in a single UTF-8 byte
     multi_byte_tokens: [
-        [".", "DOT"],
+        [".", "DOT"], // Note: DOT is here because <TODO: REASON>
```
Comment (matklad) [inline crates/ra_syntax/src/grammar.ron]: Hm, it might be just an accident actually

Comment (DJMcNab) [reply crates/ra_syntax/src/grammar.ron]: OK! Should I change it?

Comment (matklad) [reply crates/ra_syntax/src/grammar.ron]: Yep, let's try to change it it usual token and see if anything breaks!

**Pull request merged**